troposphere/static/js/components/admin/ResourceRequest.react.js

```diff
@@ -1,31 +1,98 @@
-define(function (require) {
-  "use strict";
+import React from 'react';
+import Backbone from 'backbone';
+import Router from 'react-router';
+import stores from 'stores';
+import Glyphicon from 'components/common/Glyphicon.react';
+import actions from 'actions';
+import ResourceActions from 'actions/ResourceActions';
+var RouteHandler = Router.RouteHandler;
 
-  var React = require('react/addons'),
-    Backbone = require('backbone'),
-    Router = require('react-router'),
-    stores = require('stores'),
-    Glyphicon = require('components/common/Glyphicon.react'),
-    actions = require('actions'),
-    RouteHandler = Router.RouteHandler,
-    ResourceActions = require('actions/ResourceActions');
-
-  return React.createClass({
+var ResourceRequest = React.createClass({
 
     mixins: [Router.State],
 
     getInitialState: function(){
       return {
-        AUSearch: 0,
+        AUSearch: null,
+        currentAllocation: null,
+        currentQuota: null,
+        quotaSearch: null,
         delta: 525600,
         expire: true,
         response: "",
+        request: null,
         canSubmit: false,
-        newAllocationId: null,
         displayAdmin: false
       };
     },
 
+    componentWillReceiveProps: function(){
```

Comment:
I'd like some input on this. The reasoning is that when a new request is clicked from the left panel after a `ResourceRequest` is mounted, that previous request is not unmounted before rendering a request with the newly selected ID. 
Since `updateState()` only updates state variables if they don't already exist, this means that the newly selected request will not render unless state is set back to its initial value.